src/wallet/rpcwallet.cpp

```diff
@@ -3460,6 +3532,7 @@ UniValue gettransaction(const UniValue& params, bool fHelp)
         nFee = -(wtx.getTxBase()->GetFeeAmount(nDebit) + cswInTotAmount);
     }
 
+    entry.pushKV("version", wtx.getTxBase()->nVersion);
```

Comment:
Put this line inside `WalletTxToJSON` method.

Reply:
check where is WalletTxToJSON is called